constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java

```diff
@@ -24,45 +24,58 @@
 import static org.jboss.pnc.enums.BuildProgress.PENDING;
 
 /**
- * Status of a running or finished build.
+ * Status of a running/completed Build/GroupBuild.
+ * 
+ * The status class is shared by both Build and GroupBuild dtos, which are published through UI/WebSocket.
  *
  * @author Jakub Bartecek &lt;jbartece@redhat.com&gt;
  *
  */
 public enum BuildStatus {
     /**
-     * Build completed successfully.
+     * Build/GroupBuild has completed successfully. The dependant Builds can be scheduled.
      */
     SUCCESS(FINISHED, true),
 
     /**
-     * Build failed.
+     * Build/GroupBuild has failed. The status is propagated through dependants as REJECTED_FAILED_DEPENDENCIES.
+     *
+     * A GroupBuild is reported FAILED with is at least one underlying Build with the final status of (FAILED, REJECTED,
+     * SYSTEM_ERROR)
      */
     FAILED(FINISHED, false),
 
     /**
-     * A build has been requested (possibly via dependencies) but no actual build happened as it was not required (no
-     * updates).
+     * A Build has been requested (possibly via dependencies) but no actual build happened as it was not required (no
```

Comment:
dependencies? I think it's dependants.
But do we require new build of a dependency when it doesn't require rebuild? Why and in which cases?

Reply:
Good catch, it should be dependants. 

We require a rebuild of dependency if its BC has changed or if dependency's dependency needs rebuild(and so on). How you define which dependencies you are checking relies on IMPLICIT/EXPLICIT parameter.

Reply:
Yeah, but you say "A Build has been requested (possibly via dependants)". Do we request dependecies to build even when they don't require rebuild?

Reply:
Yes. If you request a build, it goes through explicit dependencies. The dependency is requested, but it finishes immediately with NO_REBUILD_REQUIRED(REJECTED_ALREADY_BUILT on prod) if its BC was not modified.

Reply:
I don't like that behaviour, but fine :)